certbot/plugins/manual.py

```diff
@@ -59,203 +52,113 @@ class Authenticator(common.Plugin):
 If you don't have HTTP server configured, you can run the following
 command on the target server (as root):
 
-{command}
-"""}
-
-    # a disclaimer about your current IP being transmitted to Let's Encrypt's servers.
-    IP_DISCLAIMER = """\
-NOTE: The IP of this machine will be publicly logged as having requested this certificate. \
-If you're running certbot in manual mode on a machine that is not your server, \
-please ensure you're okay with that.
-
-Are you OK with your IP being logged?
-"""
-
-    # "cd /tmp/certbot" makes sure user doesn't serve /root,
-    # separate "public_html" ensures that cert.pem/key.pem are not
-    # served and makes it more obvious that Python command will serve
-    # anything recursively under the cwd
-
-    CMD_TEMPLATE = """\
-mkdir -p {root}/public_html/{achall.URI_ROOT_PATH}
-cd {root}/public_html
+mkdir -p /tmp/certbot/public_html/{achall.URI_ROOT_PATH}
+cd /tmp/certbot/public_html
 printf "%s" {validation} > {achall.URI_ROOT_PATH}/{encoded_token}
 # run only once per server:
 $(command -v python2 || command -v python2.7 || command -v python2.6) -c \\
 "import BaseHTTPServer, SimpleHTTPServer; \\
 s = BaseHTTPServer.HTTPServer(('', {port}), SimpleHTTPServer.SimpleHTTPRequestHandler); \\
 s.serve_forever()" """
-    """Command template."""
 
     def __init__(self, *args, **kwargs):
         super(Authenticator, self).__init__(*args, **kwargs)
-        self._root = (tempfile.mkdtemp() if self.conf("test-mode")
-                      else "/tmp/certbot")
-        self._httpd = None
+        self.env = dict()
 
     @classmethod
     def add_parser_arguments(cls, add):
-        add("test-mode", action="store_true",
-            help="Test mode. Executes the manual command in subprocess.")
-        add("public-ip-logging-ok", action="store_true",
-            help="Automatically allows public IP logging.")
-
-    def prepare(self):  # pylint: disable=missing-docstring,no-self-use
-        if self.config.noninteractive_mode and not self.conf("test-mode"):
-            raise errors.PluginError("Running manual mode non-interactively is not supported")
+        add('auth-hook',
```

Comment:
According to your comment [here](https://github.com/certbot/certbot/issues/3767#issue-187879645):
> We disagreed on whether --auth-hook should always be run or whether it should be specific to --manual, but concluded that this would be the prerogative of whoever implements this refactor.

Both Joona and I argued against doing this in #3767.